clang/lib/Frontend/CompileJobCache.cpp

```diff
@@ -0,0 +1,981 @@
+//===- CompileJobCache.cpp ------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "clang/Frontend/CompileJobCache.h"
+#include "CachedDiagnostics.h"
+#include "clang/Basic/DiagnosticCAS.h"
+#include "clang/Frontend/CASDependencyCollector.h"
+#include "clang/Frontend/CompileJobCacheKey.h"
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Lex/PreprocessorOptions.h"
+#include "llvm/CAS/ActionCache.h"
+#include "llvm/CAS/CASOutputBackend.h"
+#include "llvm/CASUtil/Utils.h"
+#include "llvm/MCCAS/MCCASObjectV1.h"
+#include "llvm/RemoteCachingService/Client.h"
```

Comment:
I think we need to add `MCCAS` and `RemoteCachingService` as a target dependencies in `CMakeLists.txt`. We might also be able to remove those targets from `tools/driver/CMakeLists.txt ` now?

Reply:
Done 👍